### ISSUE TITLE
Failure to parse alternate macro syntax

### ISSUE THREAD

**Issue by CohenArthur** [Rust-GCC/gccrs#1578]
I tried this code:

```rust
pub macro test($item:item) {
    /* compiler built-in */
}
```

I expected to see this happen: no error, provided that the `decl_macro` feature is enabled

Instead, this happened: 
```rust
test.rs:3:5: error: failed to parse item in crate
    3 | pub macro test($item:item) {
      |     ^
```

This is important since `libcore 1.49` relies on this syntax for some of the macros, such as `#[test]` or `#[bench]`

**Comment by CohenArthur** [comment]
This was fixed by #1708

**Comment by CohenArthur** [comment]
reopening this issue as decl macros are not handled properly within blocks:
```rust
#![feature(decl_macro)]

fn main() {
    macro_rules! foo {
        () => {};
    }

    macro bar() {
        () => {};
    }
}
```
is valid Rust code but fails with the following error:
```rust
test.rs:8:5: error: found unexpected token ‘macro’ in null denotation
    8 |     macro bar() {
      |     ^~~~~
test.rs:8:5: error: failed to parse expression for expression without block (pratt-parsed expression is null)
test.rs:8:5: error: failed to parse statement or expression without block in block expression
test.rs:8:11: error: unrecognised token ‘identifier’ for start of item
    8 |     macro bar() {
      |           ^~~
test.rs:8:11: error: failed to parse item in crate
```

commit: f990f89e90a
